db-client-java/src/main/java/com/eventstore/dbclient/ReadSubscription.java

```diff
@@ -64,10 +59,31 @@ public void request(long n) {
         if (n <= 0) {
             subscriber.onError(new IllegalArgumentException("non-positive subscription request: " + n));
         }
-        lock.lock();
-        requested.updateAndGet(current -> current + n);
-        hasRequested.signal();
-        lock.unlock();
+
+        request0(n);
+    }
+
+    private void request0(long n)
+    {
+        long bufferRequestSize = 512*3/4;
```

Comment:
Yes, it should definitely be configurable with a reasonable default. And I think I actually got it wrong, it should not be 3/4 it should be 1/4, the idea being that once 3/4 of all requests have been sent you want to preemptively send more if available so that on the server that counter never reaches zero, even for a short while.